### ISSUE TITLE
Bug: [no-deprecated] potential false positive

### ISSUE THREAD

**Issue by jits** [typescript-eslint/typescript-eslint#10155]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.5.2&fileType=.ts&code=PQKhAIAEBMFMAcBOsDGBDALra4TAFABmArgHYoYCWA9qeAGLXUAUAHgFzgDOGilpAcwCU4AN75w4ZBmKI6rANz4Avvnyk0AW1hd4aFLAZMxE8CXJVa4AEZpEzEeMmTpsugHJbid0smrV%2BIzUAHReDvhAA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
/** @deprecated */
function Foo(x: string) {
  return x;
}

namespace Foo {
  function bar() {
    return 'bar';
  }
}

Foo.bar() // This flags an `eslint@typescript-eslint/no-deprecated` error
```

### ESLint Config

```javascript
{
  "rules": {}
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

I expected that no `eslint@typescript-eslint/no-deprecated` error is raised.

### Actual Result

An `eslint@typescript-eslint/no-deprecated` is raised

### Additional Info

The actual issue I'm getting is when using the [ng-mocks](https://github.com/help-me-mom/ng-mocks) library, when calling `MockInstance.scope()`, which I've traced back to https://github.com/help-me-mom/ng-mocks/blob/6f36a1a29810b4c54a62ff5f334631278b49b834/libs/ng-mocks/src/lib/mock-instance/mock-instance.ts#L282 (just in case it helps).

**Comment by bradzacher** [comment]
This is working as expected because as far as TS is concerned the declarations are merged into a single declaration.

If one is deprecated they both are.

You can even see this in the TS decorations - TS itself strikes through the name and we report on the same location!

![Screenshot_20241016-213414.png](https://github.com/user-attachments/assets/b938f422-0ae7-4008-901d-11f42396e65d)

**Comment by jits** [comment]
Thanks @bradzacher. I got this when upgrading from `^8.3.0` -> `^8.9.0`, so wasn't sure if it was a bug or working as intended. I'll report this on the ng-mocks repo.
